tools/src/main/scala/scala/scalanative/linker/ClassPath.scala

```diff
@@ -50,5 +54,13 @@ object ClassPath {
           deserializeBinary(directory.read(file))
         }
       })
+
+    def classesWithEntryPoints(): Iterable[Global.Top] = {
+      files.filter {
+        case (top, file) =>
+          val buffer = directory.read(file, len = 1)
+          buffer.get != 0
```

Comment:
```suggestion
          buffer.get() != 0
```
`get()` is side-effecting (calling it twice will not return the same thing), so it should always be called with explicit `()`. This is a commonly used convention in Scala.

Reply:
This information should really be cached. Otherwise all files will have to reread on every linking request.